src/Uno.UI/UI/Xaml/Controls/Grid/Grid.Layout.cs

```diff
@@ -980,9 +980,17 @@ private static double GetAvailableSizeForPosition(Memory<double> calculatedPixel
 		/// </summary>
 		private static double GetAdjustmentForSpacing(int rowOrColumnSpan, double spacing) => spacing * (rowOrColumnSpan - 1);
 
-		private static Memory<ViewPosition> FindStarSizeChildren(Span<ViewPosition> positions, Memory<ViewPosition> pixelSizeChildren, List<ViewPosition> autoSizeChildren)
+		private ViewPosition[] _resCache;
+
+		private Memory<ViewPosition> FindStarSizeChildren(Span<ViewPosition> positions, Memory<ViewPosition> pixelSizeChildren, List<ViewPosition> autoSizeChildren)
 		{
-			var res = new Memory<ViewPosition>(new ViewPosition[positions.Length]);
+			if ((_resCache?.Length ?? -1) != positions.Length)
```

Comment:
I'm wondering if we should use a ArrayPool instead.

Reply:
I used both: I really expect the `_resCache` to never change during the lifetime of a `Grid`, so we can avoid the cost of lookup in the pool.